test/Sema/diag_unintended_optional_behavior.swift

```diff
@@ -7,76 +7,44 @@ func takeAny(_ left: Any, _ right: Any) -> Int? {
 func throwing() throws -> Int? {}
 
 func warnOptionalToAnyCoercion(value x: Int?) -> Any {
-  let a: Any = x // expected-warning {{expression implicitly coerced from 'Int?' to 'Any'}}
```

Comment:
We should be emitting a warning here as `x` isn't an IUO. Let me take a look at what's going on!